### ISSUE TITLE
deck.gl Scatterplot JAVASCRIPT TOOLTIP GENERATOR tooltip no longer working in version 1.4.0 and above

### ISSUE THREAD

**Issue by xtengx** [apache/superset#18696]
#### How to reproduce the bug

1. Create deck.gl scatter plot
2. Create JAVASCRIPT TOOLTIP GENERATOR tooltip i.e. d => `TEST</br>`;
3. Hover over dot in plot

### Expected results

Tooltip should show "TEST"

### Actual results

"Unexpected error
An error occurred while rendering the visualization: Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=TypeError%3A%20r(...).runInNewContext%20is%20not%20a%20function&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

See more"

#### Screenshots

![image](https://user-images.githubusercontent.com/34407987/153762437-43b38acc-79b0-43cb-8963-a39aa5d18f60.png)

### Environment

(please complete the following information):

- browser type and version:
- superset version: `superset version`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by villebro** [comment]
I was able to reproduce this problem. I also noticed that the Path example chart is broken, but works if the "JAVASCRIPT DATA INTERCEPTOR" is disabled.

Original:
<img width="1477" alt="image" src="https://user-images.githubusercontent.com/33317356/154260451-27e4edcc-d3f5-4004-8f37-b79e41134915.png">

With removed interceptor:
<img width="1478" alt="image" src="https://user-images.githubusercontent.com/33317356/154260565-161e5b3e-6d41-4bf7-95bd-076919784198.png">

**Comment by chanyou0311** [comment]
I have same problem. It seems to be caused by upgrading webpack to version 5 without resolving configurations.
I've create PR. I hope to fix your problem by this changes!
https://github.com/apache/superset/pull/19059

**Comment by rusackas** [comment]
Since the PR is merged and the thread has gone quiet for upward of a year, I'll optimistically mark this as closed. Happy to re-open if this is still an issue in 2.0.x

**Comment by DataMinerR** [comment]
Hello @rusackas ,

this IS unfortunately still an issue in 2.1.0. I get the same error. 

It would be great, if this could be solved. Unfortunately, my technical knowledge is not enough to do it myself.

BR

**Comment by DataMinerR** [comment]
Dear @rusackas ,

do we have any update on this issue?

Thank you!

**Comment by rusackas** [comment]
> do we have any update on this issue?

Nope... based on the PRs merged in this thread, I thought it was a solved problem. If it's an issue still, I think we should open up a fresh Issue including repro steps, displayed errors, and known incompatible versions.

On the other hand, this feature is questionable in terms of security - I have half a mind to deprecate it and replace it entirely with something new. If anyone has an interest in that effort, we should talk on Slack.

**Comment by DataMinerR** [comment]
Hello @rusackas ,

thank you for the update!

I wonder if there is any way to turn the tooltip completely off? Because right now it is showing the wrong thing when hovered over the map...

**Comment by DataMinerR** [comment]
Hello @rusackas ,

would you be so kind and have a look at my question at stackoverflow?

https://stackoverflow.com/questions/77404016/turn-off-deck-gl-tooltip-in-superset


Thanks & BR

**Comment by JohnDietrich-Pepper** [comment]
This is still an issue in 3.0 .  Really need some way to customize tooltips...

**Comment by rusackas** [comment]
This feature is a known XSS concern, and is likely due for deprecation, honestly. I intend to propose that for 5.0 (having missed the 4.0 proposals window). I think we should move to a different means of templatizing tooltips (e.g. handlebars/jinja) that can be applicable to more chart types, including replacing these problematic ones for DeckGL. If anyone has an interest in pursuing this SIP, please join SIP Office Hours or write one up!

**Comment by vin01** [comment]
I believe (not-a-)problem here is the [CSP policy from Talisman](https://github.com/apache/superset/blob/master/superset/config.py#L1447) which does not allow `unsafe-eval` which is required for this feature and gets blocked by browser

following talisman config works:

```
TALISMAN_CONFIG = {
    "content_security_policy": {
        "base-uri": ["'self'"],
        "default-src": ["'self'"],
        "img-src": [
            "'self'",
            "blob:",
            "data:",
            "https://apachesuperset.gateway.scarf.sh",
            "https://static.scarf.sh/",
        ],
        "worker-src": ["'self'", "blob:"],
        "connect-src": [
            "'self'",
            "https://api.mapbox.com",
            "https://events.mapbox.com",
        ],
        "object-src": "'none'",
        "style-src": [
            "'self'",
            "'unsafe-inline'",
        ],
        "script-src": ["'self'", "'strict-dynamic'", "'unsafe-eval'"],
    },
    "content_security_policy_nonce_in": ["script-src"],
    "force_https": False,
    "session_cookie_secure": False,
}
```

Given the nature of this issue, it would be ideal to have an alternative way to templatize tooltips. However in some cases this could be acceptable risk where users can be trusted with customized JS, we could document it, at least until a better way is possible.

**Comment by rusackas** [comment]
Based on the last comment and the fact that this hasn't been touched in upward of a year, I'll close it (again). Thanks for the help.

I still intend to remove this (but I didn't for 5.0) and replace it with something better... perhaps Handlebars-based.
